### PR TITLE
systemd-networkd: Update to v249

### LINKED ISSUE #50: Upgrade systemd to v248 or v249 on Gateways
Packages for Ubuntu 20.04 can be found here:
https://code.launchpad.net/~ubuntu-support-team/+archive/ubuntu/systemd/+build/21808586

I tested the deploy on ft01.ov.ffmuc.net which worked flawless. All required packages are in the /root/ dir of that machine.

Most important before an upgrade of the gateways is performed is to make sure all networkd configs use the new way of configuring BATMAN and also make sure to get the FDB entries right:
https://github.com/systemd/systemd/pull/19639/files

### PR DESCRIPTION
**Pull request by krombel**
fixes #50

### CODE REVIEW

File: systemd-networkd/files/systemd-netdev.jinja2 [hunk 1]
```diff
@@ -37,22 +36,15 @@ Endpoint={{ node | regex_replace('in\.ffmuc\.net','ext.ffmuc.net') }}:{{ wiregua
 {%- set current_site = interface | regex_replace('vx-','') %}
 {%- for node,bla in salt['mine.get']('netbox:role:name:nextgen-gateway', 'minion_id', tgt_type='pillar').items() | sort %}
 {%- if grains['id'] not in node %}
-{%- set node_location = salt['mine.get'](node, 'minion_location', tgt_type='glob') %}
-{%- set wireguard_public_key = salt['mine.get'](node,'minion_wireguard_public', tgt_type='glob')[node]  %}
-{%- if wireguard_public_key %}
-{%- set link_local = salt['wireguard_v6.generate'](wireguard_public_key) %}
 [VXLAN]
 VNI={{ sites[current_site]['vni'] }}
-Local={{ salt['wireguard_v6.generate'](salt['pillar.get']('netbox:config_context:wireguard:public_key')) }}
```
Comment (awlx) [inline systemd-networkd/files/systemd-netdev.jinja2:46]: I think local-ip needs to bet set.

File: systemd-networkd/init.sls [hunk 2]
```diff
@@ -1,3 +1,20 @@
+
+
+{% set systemd_version = "249.164.gf571d9d5f0+20.04.20210724133612" %}
```
Comment (awlx) [inline systemd-networkd/init.sls:3]: Can we only do this if it's a gateway?

**Review by awlx — CHANGES_REQUESTED** [review]

**Comment by awlx** [comment]
Nice :). Left some comments.

**Review by awlx — DISMISSED** [review]
LGTM

File: systemd-networkd/files/batadv-throughput.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+for i in `ip -br a | grep vx- | awk '{print $1}'`; do 
+    echo Set throughput_override on $i
```
Comment (awlx) [inline systemd-networkd/files/batadv-throughput.sh:4]: would add "" here.

**Review by awlx — CHANGES_REQUESTED** [review]

File: systemd-networkd/files/batadv-throughput.sh (hunk 3 above)
Comment (awlx) [reply systemd-networkd/files/batadv-throughput.sh]:
```suggestion
    logger -t batadv-override  "Set throughput_override on $i"
```

File: systemd-networkd/files/batadv-throughput.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+for i in `ip -br a | grep vx- | awk '{print $1}'`; do 
+    echo "Set throughput_override on $i"
```
Comment (awlx) [inline systemd-networkd/files/batadv-throughput.sh:4]:
```suggestion
    logger -t batadv-override "Set throughput_override on $i"
```

**Review by awlx — CHANGES_REQUESTED** [review]

**Review by awlx — APPROVED** [review]

File: systemd-networkd/files/systemd-network.jinja2 [hunk 5]
```diff
@@ -61,4 +61,23 @@ Bridge={{ interface | regex_replace('bat-', 'br-') }}
...
+{%- set sites = salt['pillar.get']('netbox:config_context:site_config') %}
+{%- set current_site = interface | regex_replace('vx-','') %}
+{%- for node,bla in salt['mine.get']('netbox:role:name:nextgen-gateway', 'minion_id', tgt_type='pillar').items() | sort %}
+{%- if grains['id'] not in node %}
+{%- set wireguard_public_key = salt['mine.get'](node,'minion_wireguard_public', tgt_type='glob')[node]  %}
+{%- if wireguard_public_key %}
+{%- set link_local = salt['wireguard_v6.generate'](wireguard_public_key) %}
+
+[BridgeFDB]
+MACAddress=00:00:00:00:00
```
Comment (awlx) [inline systemd-networkd/files/systemd-network.jinja2:76]:
```suggestion
MACAddress=00:00:00:00:00:00
```

Maybe it works with a valid mac :D. Whoops

**Comment by krombel** [comment]
now as this ist tested in production I think it is finally safe to merge it :)

**Review by awlx — APPROVED** [review]
Closes #51 as well :)

**Pull request merged**